### ISSUE TITLE
Keyboard space is not removed when interacting with password inputs (on iOS)

### ISSUE THREAD

**Issue by depyronick** [mateusz1913/react-native-avoid-softinput#149]
### Environment

Library version: 3.1.3
OS version: 14 Pro, iOS 16.2 

### Affected platforms

- [ ] Android
- [X] iOS

### Current behavior

If you focus on another input after scrolling while the keyboard is open, keyboard height space does not removed when the keyboard closed. 

### Expected behavior

Remove space.

**Comment by depyronick** [comment]
https://user-images.githubusercontent.com/13220891/218253024-aa84a0f5-6a11-4210-ad95-d5092bd51360.MOV

**Comment by mateusz1913** [comment]
Hi @depyronick , thx for sending the recording! Unfortunately I am not able to start investigation, based only on the video.

Could you paste the snippet of code which consists how you use the library for your form?

**Comment by depyronick** [comment]
@mateusz1913 

```
import {
  ScrollView
} from "react-native";

import { useCallback } from "react";

import { AvoidSoftInput } from "react-native-avoid-softinput";
import { useFocusEffect } from "@react-navigation/native";

const FUIKeyboardAwareScrollView = ({ children }) => {
  const onFocusEffect = useCallback(() => {
    AvoidSoftInput.setEnabled(true);
    AvoidSoftInput.setShouldMimicIOSBehavior(true);
    AvoidSoftInput.setEasing("linear");
    AvoidSoftInput.setHideAnimationDelay(0);
    AvoidSoftInput.setShowAnimationDelay(0);
    AvoidSoftInput.setAvoidOffset(0);

    return () => {
      AvoidSoftInput.setEnabled(false);
      AvoidSoftInput.setShouldMimicIOSBehavior(false);
    };
  }, []);

  useFocusEffect(onFocusEffect);

  return (
    <ScrollView>
      {children}
    </ScrollView>
  );
}

export {
  FUIKeyboardAwareScrollView
}
```

```
const RegisterScreen = () => {
  return (
    <FUIKeyboardAwareScrollView>
      <TextInput />
    </FUIKeyboardAwareScrollView>
  )
}
```

**Comment by depyronick** [comment]
The implementation is as simple as I provided. The problem occurs when you scroll AND focus on another input.

**Comment by mateusz1913** [comment]
Thx, are all text inputs under the same scrollview? Would you be able to provide minimal reproduction repo, so that I can run it locally and investigate?

**Comment by hirbod** [comment]
I can reproduce that applied space won't get removed once the  Keyboard was closed with the return key. 
The issue does not happen when I close the keyboard by scrolling or tapping the ScrollView (`keyboardDismissMode="on-drag"` and `keyboardShouldPersistTaps="handled"`

Here is a video demonstrating the issue.

https://user-images.githubusercontent.com/504909/220733599-b32942e3-dbc6-4fc8-9a6c-6fabcc9519cd.mp4

**Comment by mateusz1913** [comment]
👋 @hirbod thanks for recording video. Could you provide a link to minimal reproducible repository? I'm not able to start investigating the issue just with video, where I don't have a context on whether you use `AvoidSoftInput` module or `AvoidSoftInputView`

**Comment by hirbod** [comment]
@mateusz1913 I understand, I am just a bit overloaded with work right now, I try to give you an MRE, but my explanation actually nailed the repro.

I am using 

```
  useEffect(() => {
    AvoidSoftInput.setEnabled(true);
    AvoidSoftInput.setShowAnimationDelay(0);
    return () => {
      AvoidSoftInput.setEnabled(false);
    };
  }, []);
```

That's it. iOS 16, Video is from the simulator. Only reproducible when I hit the return key. 
Hope this helps. If you can't reproduce this with my informations, I'll try to make an repro soon

**Comment by mateusz1913** [comment]
Hmmm, I see that on video, you submit the form with password input with autofill - is it reproducible on a form without iOS autofill?

**Comment by mateusz1913** [comment]
I'll start with testing password inputs, but if you manage to make a repro or at least test if incorrect behavior occurs for non-password inputs, that would be massively appreciated 🙏

**Comment by hirbod** [comment]
I'll report back to you soon!

**Comment by hirbod** [comment]
@mateusz1913 I can confirm. I deleted the password input and used a normal/default input and no issues. The password input is breaking something here.

**Comment by hirbod** [comment]
Looks like OP also has the same issue, his last fields are password inputs as well.

**Comment by mateusz1913** [comment]
Got it, I'll try to look at it in the next few days, thx for your help in narrowing the issue!

**Comment by hirbod** [comment]
Thanks you!

**Comment by hirbod** [comment]
By the way, I found a workaround for now: setting the input to `textContentType="oneTimeCode"` instead of `password` or `newPassword` somehow resolved it. Looks like the extra space is somehow due to it would render a password dialog on an actual device

**Comment by mateusz1913** [comment]
👋 @hirbod  I reproduced the issue in the example repo (with new password form example) and I have a fix candidate, however, it would be good to test in "real-life" environment

Would you be able to apply following patch and check if issue is resolved?

```diff
diff --git a/packages/react-native-avoid-softinput/ios/AvoidSoftInputAnimationHandler.swift b/packages/react-native-avoid-softinput/ios/AvoidSoftInputAnimationHandler.swift
index 832a6f7..28ff603 100644
--- a/packages/react-native-avoid-softinput/ios/AvoidSoftInputAnimationHandler.swift
+++ b/packages/react-native-avoid-softinput/ios/AvoidSoftInputAnimationHandler.swift
@@ -220,6 +220,7 @@ public class AvoidSoftInputAnimationHandler: AvoidSoftInputAnimatorDelegate {
                         initialOffset: self.bottomOffset,
                         addedOffset: newBottomOffset - self.bottomOffset
                     )
+                    self.softInputVisible = true
                     self.hideAnimator.cleanup()
                 },
                 onAnimate: {
@@ -305,6 +306,7 @@ public class AvoidSoftInputAnimationHandler: AvoidSoftInputAnimatorDelegate {
                         initialOffset: 0,
                         addedOffset: self.bottomOffset
                     )
+                    self.softInputVisible = true
                     self.hideAnimator.cleanup()
                 },
                 onAnimate: {
@@ -455,6 +457,7 @@ public class AvoidSoftInputAnimationHandler: AvoidSoftInputAnimatorDelegate {
                         initialOffset: self.bottomOffset,
                         addedOffset: newBottomOffset - self.bottomOffset
                     )
+                    self.softInputVisible = true
                     self.hideAnimator.cleanup()
                 },
                 onAnimate: {
@@ -560,6 +563,7 @@ public class AvoidSoftInputAnimationHandler: AvoidSoftInputAnimatorDelegate {
                 easing: easingOption,
                 onStart: {
                     self.showAnimator.beginAnimation(initialOffset: 0, addedOffset: self.bottomOffset)
+                    self.softInputVisible = true
                     self.hideAnimator.cleanup()
                 },
                 onAnimate: {
```

**Comment by hirbod** [comment]
The patch did resolve the issue, thanks!

**Comment by depyronick** [comment]
Wow, sorry for not being able to help to fix this issue, but I am glad that it worked out and looking forward to an update.

**Comment by mateusz1913** [comment]
👋 this fix is available in v3.1.5 🚀
